gcloud-java-resourcemanager/src/main/java/com/google/gcloud/spi/ResourceManagerRpc.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.spi;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.cloudresourcemanager.model.Policy;
+import com.google.api.services.cloudresourcemanager.model.Project;
+import com.google.common.collect.ImmutableList;
+import com.google.gcloud.resourcemanager.ResourceManagerException;
+
+import java.util.Collections;
+import java.util.List;
+
+public interface ResourceManagerRpc {
+
+  public enum Permission {
+    CREATE,
+    DELETE,
+    GET,
+    LIST,
+    UPDATE,
+    GET_IAM_POLICY,
+    SET_IAM_POLICY
+  }
+
+  public enum DeleteResult {
+    SUCCESS,
+    ALREADY_DELETED
+  }
+
+  public enum UndeleteResult {
+    SUCCESS,
+    DELETE_IN_PROGRESS,
+    GONE
+  }
+
+  public enum SetIamPolicyResult {
+    SUCCESS,
+    ETAG_NOT_MATCH
```

Comment:
I just tested this using the API explorer, and I realized that I misinterpreted the IAM documentation.  (They say "When you make a request to Cloud IAM with an etag value, Cloud IAM compares the etag value in the request with the existing etag value associated with the policy," I thought the "with an etag value" belonged to the "request," not the "Cloud IAM.")  

To clarify: the policy seems to be given an etag value at the very beginning (most of the time).  The etag value is always checked, and the user should always call `getIamPolicy`, get the etag value on that policy, and use it in their setIamPolicy request.  

I couldn't find reproduce this, but it seems the policy may sometimes not have an etag value. (The documentation says "Use the etag value when setting the policy only if the corresponding policy in GetPolicyResponse contains an etag value.")  In these cases, the etag value won't be compared.